relay-server/src/actors/project_upstream.rs

```diff
@@ -224,6 +232,33 @@ impl UpstreamProjectSource {
                     public_keys: channels_batch.keys().copied().collect(),
                     full_config: self.config.processing_enabled(),
                     no_cache: channels_batch.values().any(|c| c.no_cache),
+                    version: if channels_batch.values().any(|c| c.almost_expired()) {
```

Comment:
I worry that this will mean we'll effectively ask very few requests to v3.  Maybe the batching selection we should make the selection `channel.no_cache || channel.almost_expired` since both no_cache and v2 requests will be handled as v2 in the backend.... and oh no, sidetrack, did we break no_cache in the backend?sometime?

Reply:
Split into three batches now.